### PR TITLE
Fix loader

### PR DESCRIPTION
**Pull request by InvisibleSymbol**
`response` wasn't being checked, array operation caused an `KeyError` when an invalid key was specified. Now raises an correct `InvalidKeyException`.

### CODE REVIEW

File: circleguard/loadable.py [hunk 1]
```diff
@@ -663,6 +663,7 @@ class ReplayPath(Replay):
     def __init__(self, path, cache=None):
         self.log = logging.getLogger(__name__ + ".ReplayPath")
         self.path = path
+        self.hash = None
```
Comment (tybug) [inline circleguard/loadable.py]: is this hash being used for anything?

File: tests/test.py [hunk 2]
```diff
@@ -203,6 +204,47 @@ def test_map_user_slice(self):
         # test slicing
         self.assertListEqual([r.map_id for r in self.mu[0:2]], [795627, 795627])
 
+
+class TestLoader(CGTestCase):
```
Comment (tybug) [inline tests/test.py]: combine the methods of this class; merge `test_incorrect_map_id` with `test_working_map_id` (to a method named `test_loading_map_id`, same with the three `user_id` methods and the two `username` methods.

File: tests/test.py [hunk 3]
```diff
@@ -203,6 +204,47 @@ def test_map_user_slice(self):
...
+    def test_incorrect_username(self):
+        result = self.loader.username(0)
+        self.assertEqual(result, "")
+
+    def test_working_username(self):
+        result = self.loader.username(13506780)
+        self.assertEqual(result, "] [")
+
+    def test_incorrect_key(self):
+        loader = Loader("LULW 727 LULW")
```
Comment (tybug) [inline tests/test.py]: use a more professional key please :p `"incorrect key"` works

File: circleguard/loadable.py (hunk 1 above)
Comment (InvisibleSymbol) [reply circleguard/loadable.py]: If the api fails this is the only way to Identify a map. Same as how the username is kept even if we get the user id.

**Pull request merged**